misk-hibernate/src/test/kotlin/misk/hibernate/TransformedColumnTest.kt

```diff
@@ -0,0 +1,451 @@
+package misk.hibernate
+
+import misk.MiskTestingServiceModule
+import misk.config.Config
+import misk.config.MiskConfig
+import misk.environment.Environment
+import misk.environment.EnvironmentModule
+import misk.inject.KAbstractModule
+import misk.jdbc.DataSourceConfig
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.data.Percentage
+import org.junit.jupiter.api.Test
+import java.io.Serializable
+import java.util.Objects
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.persistence.AttributeConverter
+import javax.persistence.Column
+import javax.persistence.Convert
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.Table
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.jvm.javaField
+
+interface SwappableTransformer {
+  fun assemble(ctx: TransformerContext, owner: Any?, value: Serializable): Any
+
+  fun disassemble(ctx: TransformerContext, value: Any): Serializable
+}
+
+class DelegatingTransformer(private val ctx: TransformerContext) : Transformer(ctx) {
+  @Inject
+  lateinit var transformer: SwappableTransformer
+
+  override fun assemble(owner: Any?, value: Serializable): Any = transformer.assemble(ctx, owner, value)
+  override fun disassemble(value: Any): Serializable = transformer.disassemble(ctx, value)
+}
+
+@MiskTest(startService = true)
+class TransformedColumnTest {
```

Comment:
I'm still ramping up on misk, and on top of that I've mainly used jOOQ and very little hibernate.
Between misk and hibernate, are there good facilities for observing queries made in test so that you can assert things like "exactly 1 query was made against this table," etc? That sort of assertion may be very valuable for a generic/automagic feature like this. Apologies if that's already tested for here and I just haven't recognized it yet :)

Reply:
That seems sufficient from a correctness standpoint, performance related assertions may be overkill at this point